### PR TITLE
Chore/bytes api consistency

### PR DESCRIPTION
**Pull request by Donnype**
### Changes
This PR fixes several things for Bytes:

Consistency in the API:
-  All object now have a filter endpoint.
- The filter endpoints have an offset
- The raw endpoints make more sense as well

before:
before

after:
after

I also finally phased out the awkward foreign key from normalizer_meta to boefje_meta. To ensure keeping as much data as possible, I added a data migration that should keep all raw files accessible through the API.

### Issue link
This one also fixes https://github.com/minvws/nl-kat-coordination/issues/707

It relates to https://github.com/minvws/nl-kat-coordination/issues/727.

---
### Code Checklist
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [x] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
- [x] The code does not violate Model-View-Template and our other architectural principles.
- [x] The code prioritizes readability over performance where appropriate.
- [x] The code does not bypass authentication or security mechanisms.
- [x] The code does not introduce any dependency on a library that has not been properly vetted.
- [x] The code contains docstrings, comments, and documentation where needed.

---
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make kat`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* _bullet point + screenshot (if useful) per tested functionality_

### What doesn't work:
* _bullet point + screenshot (if useful) per tested functionality_

### Bug or feature?:
* _bullet point + screenshot (if useful) if it is unclear whether something is a bug or an intended feature._

### CODE REVIEW

**Review by ammar92 — DISMISSED** [review]
Haven't tested this (yet), but looks good in general 👍 

I think the raw endpoints indeed make more sense now.

**Review by Darwinkel — APPROVED** [review]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make kat`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* No apparent regressions

File: boefjes/boefjes/clients/bytes_client.py
[the remark this answers is not in the document]
Comment (Donnype) [reply boefjes/boefjes/clients/bytes_client.py:1]: A good cleanup, but it does require some extra QA work (the client is a bit of a blind sport in the boefje test suite)

File: bytes/bytes/repositories/meta_repository.py [hunk 1]
```diff
@@ -45,26 +64,26 @@ def save_boefje_meta(self, boefje_meta: BoefjeMeta) -> None:
...
-    def get_raws(self, query_filter: RawDataFilter) -> List[RawDataMeta]:
+    def get_raw_by_id(self, raw_id: str) -> RawData:
         raise NotImplementedError()
 
-    def has_raw(self, boefje_meta: BoefjeMeta, mime_types: List[MimeType]) -> bool:
+    def get_raw(self, query_filter: RawDataFilter) -> List[RawDataMeta]:
         raise NotImplementedError()
 
-    def get_raw(self, raw_id: str) -> RawData:
+    def has_raw(self, boefje_meta: BoefjeMeta, mime_types: List[MimeType]) -> bool:
```
[the remark this answers is not in the document]
Comment (Donnype) [reply bytes/bytes/repositories/meta_repository.py:88]: This breaks some tests due to sets always trying to hash objects (also see https://github.com/pydantic/pydantic/issues/1090). The `mime_types` field in boefjes/job_models is also not a set, so I'd suggest that if we want to change this everywhere we do it in a separate issue to fix this carefully.

**Review by dekkers — CHANGES_REQUESTED** [review]

File: bytes/bytes/events/events.py [hunk 2]
```diff
@@ -20,9 +20,3 @@ class RawFileReceived(Event):
     event_id: str = "raw_file_received"
 
     raw_data: RawDataMeta
-
-
-class NormalizerMetaReceived(Event):
```
[the remark this answers is not in the document]
Comment (Donnype) [reply bytes/bytes/events/events.py:25]: I checked with JP beforehand and that code was actually already removed

File: bytes/bytes/repositories/meta_repository.py [hunk 3]
```diff
@@ -11,25 +11,44 @@ class BoefjeMetaFilter(BaseModel):
...
+    raw_id: Optional[str]
+    limit: int = 1
+    offset: int = 0
+    descending: bool = True
+
+    @root_validator(pre=False)
+    def either_organization_or_boefje_meta_id(  # pylint: disable=no-self-argument
+        cls, values: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        if values.get("organization") or values.get("raw_id"):
```
[the remark this answers is not in the document]
Comment (Donnype) [reply bytes/bytes/repositories/meta_repository.py:30]: Completely overlooked this method, thanks!

File: bytes/bytes/repositories/meta_repository.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Donnype) [reply bytes/bytes/repositories/meta_repository.py:79]: https://github.com/minvws/nl-kat-coordination/issues/796

**Review by dekkers — APPROVED** [review]

**Review by dekkers — APPROVED** [review]

**Pull request merged**